addons/point_of_sale/models/pos_order.py

```diff
@@ -615,16 +604,18 @@ def unlink(self):
 
     @api.model
     def create(self, values):
-        if values.get('session_id'):
-            # set name based on the sequence specified on the config
-            session = self.env['pos.session'].browse(values['session_id'])
-            values['name'] = session.config_id.sequence_id._next()
-            values.setdefault('pricelist_id', session.config_id.pricelist_id.id)
-        else:
-            # fallback on any pos.order sequence
-            values['name'] = self.env['ir.sequence'].next_by_code('pos.order')
+        session = self.env['pos.session'].browse(values['session_id'])
```

Comment:
session_id is not required, not sure you can do this

Reply:
yes it is https://github.com/odoo/odoo/pull/34623/commits/2d66ce66a3465e93b80ca1b927211e188aaaa715#diff-5eeb7d6a1a488dff039e5cc3300730e1R507

Reply:
oops my bad

Reply:
but do you think I should call get('session_id') and raise a ValidationError if there is no session_id? as it is done before the call to super that will trigger the error for require field?

Reply:
it comes from [there](https://github.com/odoo/odoo/commit/69cd13a0841fa81f881a616aa28424af7ce6a4c8) when session_id was not required, then it became required [here](https://github.com/odoo/odoo/commit/0e301874badac94451ec148d9deaf400a53bb0dd)

so i would write the code as if session_id is expected in values